Automattic/simplenote-android
# Issue/842 add sorting results

### Fix
Add sorting options ***Relevance***, ***Alphabetically***, ***Created***, and ***Modified*** to the search results to close #842.  The ***Relevance*** option behaves the same as the ***Modified*** option in these changes.  ***Relevance*** will be updated in a subsequent pull request.  See the screenshots and animations below for illustration.

![842_add_sorting_results_order](https://user-images.githubusercontent.com/3827611/67454100-37440a80-f5e6-11e9-8af8-c1dbdc43b0bb.png)

![842_add_sorting_results_search](https://user-images.githubusercontent.com/3827611/67454104-3a3efb00-f5e6-11e9-9bec-5f3028d63f65.png)

<a href="https://user-images.githubusercontent.com/3827611/67454110-3f03af00-f5e6-11e9-9293-29a2247f199d.gif"><img src="https://user-images.githubusercontent.com/3827611/67454110-3f03af00-f5e6-11e9-9293-29a2247f199d.gif" width="320"></a>

<a href="https://user-images.githubusercontent.com/3827611/67454112-40cd7280-f5e6-11e9-81ae-ad4bb86ecd02.gif"><img src="https://user-images.githubusercontent.com/3827611/67454112-40cd7280-f5e6-11e9-81ae-ad4bb86ecd02.gif" width="320"></a>

### Test
1. Tap ***Search*** action in top app bar.
2. Notice search sorting view shown with ***Relevance*** selected.
3. Notice search sorting direction is descending with arrow down.
4. Notice note list items have date shown with recently modified last.
5. Notice search sorting direction switch arrows at right.
6. Long-press search sorting direction switch button.
7. Notice circular ripple background touch feedback.
8. Notice ***Reverse order*** message shown.
9. Tap search sorting direction switch button.
10. Notice search sorting direction arrow rotates.
11. Notice note list items have date shown with recently modified first.
12. Tap search sorting row.
13. Notice rectangular ripple background touch feedback.
14. Notice search sorting options menu shown.
15. Tap ***Alphabetically*** option in menu.
16. Notice note list items have date hidden.
17. Notice note list items in reverse alphabetical order.
18. Tap search sorting direction switch button.
19. Notice search sorting direction arrow rotates.
20. Notice note list items have date hidden.
21. Notice note list items in alphabetical order.
22. Tap back arrow in top app bar.
23. Notice note list items return to ***Sort order*** selection in ***Settings***.

### Review
Only one developer and one designer are required to review these changes, but anyone can perform the review.

### Release
`RELEASE-NOTES.txt` was updated in d4cde887 with:
> Added search sorting by relevance, date created, date modified, and alphabetically
